src/user/service.ts

```diff
@@ -117,3 +117,92 @@ export async function validateUserCredentials(user: User, password: string, log:
   return true;
 }
 
+/**
+ * @description - Given a principal User, uses the principal id to find the UserInfo record for a user.
+ * @param user - The principal User to find the UserInfo record for.
+ * @returns The UserInfo record for the user.
+ * @throws NotFound - If the UserInfo record is not found.
+ */
+export async function findUserInfoByUser(user: User): Promise<UserInfo> {
+
+  const result = await db('user_info')
+      .select()
+      .where({principal_id: user.id})
+      .first();
+
+  if (!result) throw new NotFound(`UserInfo for user "${user.id}" not found.`);
+
+  return recordToModel(user, result);
+}
+
+/**
+ * @description - Given a principal User, uses the principal id to find the Principal user and update the UserInfo record for a user.
+ * @param user - The principal User to update the UserInfo record for.
+ * @param userInfo - new UserInfo object to update the UserInfo record with.
+ * @returns The updated UserInfo record.
+ * @throws BadRequest - If the UserInfo record is not updated.
+ */
+export async function updateUserInfo(user: User, userInfo: UserInfo): Promise<UserInfo> {
+  
+  const result = await db('user_info')
+    .where({principal_id: user.id})
+    .update(userInfo)
+    .returning('*')
```

Comment:
This unfortunately is a Postgres specific feature.

However, I don't think you have to return anything here. Most `update` functions don't, and the `put` request doesn't return its representation. So returning `<void>` is fine